src/utils/mobile_app/validation.rs

```diff
@@ -0,0 +1,63 @@
+use std::io::{Read, Seek, SeekFrom};
+use std::path::Path;
+
+use anyhow::Result;
+
+pub fn is_zip_file<R: Read + Seek>(reader: &mut R) -> Result<bool> {
```

Comment:
[nit] I find the following syntax more readable, especially for funcitons which have mutiple generics and/or trait bounds

```suggestion
pub fn is_zip_file<R>(reader: &mut R) -> Result<bool>
where
    R: Read + Seek,
{
```